### PR TITLE
Patch/120 - implement CR feedback

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**

### CODE REVIEW

**Review by dheitzer — APPROVED** [review]

File: front-end/src/app/shared/services/login.service.ts [hunk 1]
```diff
@@ -58,11 +60,15 @@ export class LoginService extends DestroyerComponent {
           window.location.href = environment.loginDotGovLogoutUrl;
         }
       }
-      this.store.dispatch(setCommitteeAccountDetailsAction({ payload: new CommitteeAccount() }));
```
Comment (Elaine-Krauss-TCG) [inline front-end/src/app/shared/services/login.service.ts:61]: Dispatching the userLoggedOutAction (or the login.gov equivalent) now triggers new resolvers in the store for committeeAccount and activeReport

File: front-end/src/app/shared/services/login.service.ts [hunk 2]
```diff
@@ -71,11 +77,6 @@ export class LoginService extends DestroyerComponent {
     this.cookieService.delete(environment.ffapiSecurityConsentCookieName);
   }
 
-  public clearUserLoggedInCookies() {
-    this.clearUserFecfileApiCookies();
-    this.cookieService.delete(environment.sessionIdCookieName);
```
Comment (Elaine-Krauss-TCG) [inline front-end/src/app/shared/services/login.service.ts:76]: This didn't work.  Apparently, the front-end is incapable of clearing cookies with the httpOnly flag set to true.

File: front-end/src/app/shared/guards/committee.guard.spec.ts [hunk 3]
```diff
@@ -24,20 +23,18 @@ describe('committeeGuard', () => {
 
   it('should return false without committee', () => {
     const router = TestBed.inject(Router);
-    const navigateSpy = spyOn(router, 'navigateByUrl').and.resolveTo(undefined);
     const route: ActivatedRouteSnapshot = {} as any; // eslint-disable-line @typescript-eslint/no-explicit-any
     const state: RouterStateSnapshot = {} as any; // eslint-disable-line @typescript-eslint/no-explicit-any
-    (executeGuard(route, state) as Observable<boolean>).subscribe((safe) => {
-      expect(safe).toBeFalse();
-      expect(navigateSpy).toHaveBeenCalled();
+    return (executeGuard(route, state) as Promise<boolean | UrlTree>).then((safe) => {
```
Comment (toddlees) [inline front-end/src/app/shared/guards/committee.guard.spec.ts:28]: nice catch!

File: front-end/src/app/store/active-report.reducer.ts [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+import { userLoggedOutAction, userLoggedOutForLoginDotGovAction } from './login.actions';
 
 export const initialState: Report | undefined = undefined;
 
 export const activeReportReducer = createReducer<Report | undefined, Action>(
   initialState,
-  on(setActiveReportAction, (_state, update) => update.payload)
+  on(setActiveReportAction, (_state, update) => update.payload),
+  on(userLoggedOutAction, () => initialState),
+  on(userLoggedOutForLoginDotGovAction, () => initialState),
```
Comment (toddlees) [inline front-end/src/app/store/active-report.reducer.ts]: this is cool i like letting the reducer opt into the action

**Review by toddlees — APPROVED** [review]

**Pull request merged**